mpas_analysis/ocean/meridional_overturning_circulation.py

```diff
@@ -506,41 +483,38 @@ def write_file(dsMOCTimeSeries):
 #     return (mocDictClimo, mocDictTseries)
 
 
-def _compute_transport(maxEdgesInTransect, transectEdgeGlobalIDs,
-                       transectEdgeMaskSigns, nz, dvEdge, refLayerThickness,
+def _compute_transport(transectEdgeMaskSigns, nz, dvEdge, refLayerThickness,
                        horizontalVel):  # {{{
 
     '''compute mass transport across southern transect of ocean basin'''
 
-    transportZEdge = np.zeros([nz, maxEdgesInTransect])
-    for i in range(maxEdgesInTransect):
-        if transectEdgeGlobalIDs[i] == 0:
-            break
-        # subtract 1 because of python 0-indexing
-        iEdge = transectEdgeGlobalIDs[i] - 1
-        transportZEdge[:, i] = horizontalVel[iEdge, :] * \
-            transectEdgeMaskSigns[iEdge, np.newaxis] * \
-            dvEdge[iEdge, np.newaxis] * \
-            refLayerThickness[np.newaxis, :]
-    transportZ = transportZEdge.sum(axis=1)
+    transect = transectEdgeMaskSigns != 0
+    transportZ = (refLayerThickness*horizontalVel.where(transect)
+                  *transectEdgeMaskSigns.where(transect)
+                  *dvEdge.where(transect)
+                  ).sum('nEdges')
```

Comment:
@pwolfram, very nice!  I had seen that this could be vectorized and I appreciate you doing it.